### PR TITLE
Subscribers can get access to write Backyard Conspiracy articles

### PR DESCRIPTION
**Pull request by DavDan1**
PT: https://www.pivotaltracker.com/story/show/178177459

### User Story
```
As a subscriber
In order to contribute to this amazing news website
I'd like to be able to write Backyard Conspiracies
```
### Changes Suggested
- Check if subscriber - if YES render a modal button that opens up a create modal
- Display success message if the article is pubished
- Add sad path: no geolocation
- Add sad path for no articles in area

### What we have learned working on this feature
- Work with location
- Work with Redux
 
Screenshot 2021-05-23 at 22 53 27

### CODE REVIEW

**Review by arunbhalli — APPROVED** [review]

File: cypress/integration/subscriberCanWriteBackyardArticle.feature.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          );
+        },
+      });
+      cy.get('[data-cy=navbar]').within(() => {
+        cy.get('[data-cy=backyard-tab]').click();
+      });
+      cy.window().its('store').invoke('dispatch', {
+        type: 'AUTHENTICATE',
+        payload: 'Welcome back Bob!',
+      });
```
Comment (LucasKnudsen) [inline cypress/integration/subscriberCanWriteBackyardArticle.feature.js:39]: Nice

File: cypress/integration/subscriberCanWriteBackyardArticle.feature.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            };
+
+            cy.stub(
+              window.navigator.geolocation,
+              'getCurrentPosition'
+            ).callsFake((callback) => {
+              return callback(stubLocation);
+            });
+          },
+        });
```
Comment (LucasKnudsen) [inline cypress/integration/subscriberCanWriteBackyardArticle.feature.js]: Could this be moved out into the outer 'unsuccessful' describe?

File: cypress/integration/visitorCanAccessBackyard.feature.js [hunk 3]
```diff
@@ -59,7 +62,40 @@ describe('visitor can access Backyard site', () => {
...
+      cy.get('[data-cy=navbar]').within(() => {
+        cy.get('[data-cy=backyard-tab]').click();
+      });
+    });
+
+    it('is expected to show an message if there are no articles', () => {
+      cy.get('[data-cy=popup-message]').should(
+        'contain',
+        'There are no articles available in your area'
+      );
```
Comment (LucasKnudsen) [inline cypress/integration/visitorCanAccessBackyard.feature.js:94]: API is now up to date with a location object even tho articles are empty 👍

File: src/components/BackyardDashboard.jsx [hunk 4]
```diff
@@ -58,6 +59,7 @@ const BackyardDashboard = () => {
                 ? `Backyard Conspiracies from ${backyardArticles[0].location}`
                 : 'Allow your location!'}
             </h2>
+            {(subscriber && backyardArticles[0]) && <BackyardForm location={backyardArticles[0].location} />}
           </Segment>
```
Comment (LucasKnudsen) [inline src/components/BackyardDashboard.jsx]: 😆

Comment (LucasKnudsen) [reply src/components/BackyardDashboard.jsx]: Shouldn't he be able to create an article even tho there are none? :P

File: src/components/BackyardForm.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { Modal, Button, Form } from 'semantic-ui-react';
+import BackyardArticles from '../modules/BackyardArticles';
+
+const BackyardForm = (props) => {
+  const [open, setOpen] = useState(false);
+
```
Comment (LucasKnudsen) [inline src/components/BackyardForm.jsx]: Could be destructured into { location }

File: src/modules/BackyardArticles.js [hunk 6]
```diff
@@ -8,12 +8,19 @@ const BackyardArticles = {
...
-        store.dispatch({
-          type: 'SET_BACKYARD_ARTICLES',
-          payload: response.data.backyardArticles,
-        });
+        if (response.data.backyard_articles[0]) {
+          store.dispatch({
+            type: 'SET_BACKYARD_ARTICLES',
+            payload: response.data.backyard_articles,
+          });
+        } else {
```
Comment (LucasKnudsen) [inline src/modules/BackyardArticles.js]: Also up for refactoring now with the new location object

**Review by LucasKnudsen — APPROVED** [review]

**Review by LucasKnudsen — APPROVED** [review]

**Review by LucasKnudsen — APPROVED** [review]

**Pull request merged**